crates/starknet-devnet-core/src/blocks/mod.rs

```diff
@@ -27,6 +27,7 @@ pub(crate) struct StarknetBlocks {
     pub(crate) hash_to_state_diff: HashMap<BlockHash, StateDiff>,
     pub(crate) hash_to_state: HashMap<BlockHash, StarknetState>,
     pub(crate) aborted_blocks: Vec<Felt>,
+    pub(crate) starting_block_number: u64,
```

Comment:
why do we need to introduce `starting_block_number` seems that it's just `DEVNET_DEFAULT_STARTING_BLOCK_NUMBER` and it never gets updated

Reply:
it's not in case of forking, and it's later needed in block abortion; see https://github.com/0xSpaceShard/starknet-devnet-rs/pull/499#discussion_r1656858900